client/pkg/logutil/zap.go

```diff
@@ -33,6 +34,39 @@ func CreateDefaultZapLogger(level zapcore.Level) (*zap.Logger, error) {
 	return c, nil
 }
 
+// CreateUtilZapLogger creates a logger with default zap configuration can redirect log to /dev/null
+func CreateUtilZapLogger(level zapcore.Level) *zap.Logger {
```

Comment:
I was able to keep returning the `*zap.Logger` and an `error`, by writing it like so:

```go
func CreateUtilZapLogger(level zapcore.Level) (*zap.Logger, error) {
	lcfg := DefaultZapLoggerConfig
	lcfg.Level = zap.NewAtomicLevelAt(level)
	infoLevel := zap.LevelEnablerFunc(func(level zapcore.Level) bool {
		return level <= zapcore.InfoLevel
	})
	errorFatalLevel := zap.LevelEnablerFunc(func(level zapcore.Level) bool {
		return level > zapcore.InfoLevel
	})
	stdoutSyncer := zapcore.Lock(os.Stdout)
	stderrSyncer := zapcore.Lock(os.Stderr)
	opts := []zap.Option{
		zap.WrapCore(func(core zapcore.Core) zapcore.Core {
			return zapcore.NewTee(
				zapcore.NewCore(
					zapcore.NewJSONEncoder(lcfg.EncoderConfig),
					stdoutSyncer,
					infoLevel,
				),
				zapcore.NewCore(
					zapcore.NewJSONEncoder(lcfg.EncoderConfig),
					stderrSyncer,
					errorFatalLevel,
				),
			)
		}),
	}
	return lcfg.Build(opts...)
}
```


Which also makes me think if this should be an extension to `CreateDefaultZapLogger()`, and allow it to receive `zap.Option`s.

Reply:
I'm not too fond of the name `CreateUtilZapLogger`. But I don't want to request more changes, being respectful of the contributor's time, as we don't have a conclusion on the parent issue :slightly_smiling_face:.